source/MaterialXGenShader/OpenColorIOManagementSystem.h

```diff
@@ -0,0 +1,81 @@
+//
+// Copyright Contributors to the MaterialX Project
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#ifndef MATERIALX_OCIO_COLOR_MANAGEMENT_SYSTEM_H
+#define MATERIALX_OCIO_COLOR_MANAGEMENT_SYSTEM_H
+
+#ifdef MATERIALX_BUILD_OCIO
+/// @file
+/// OCIO color management system implementation
+
+#include <MaterialXCore/Definition.h>
+
+#include <MaterialXGenShader/DefaultColorManagementSystem.h>
+#include <OpenColorIO/OpenColorABI.h>
+#include <OpenColorIO/OpenColorIO.h>
+
+#include <OpenColorIO/OpenColorTypes.h>
+#include <map>
+#include <memory>
+
+namespace OCIO = OCIO_NAMESPACE;
+
+MATERIALX_NAMESPACE_BEGIN
+
+/// A shared pointer to a OpenColorIOManagementSystem
+using OpenColorIOManagementSystemPtr = std::shared_ptr<class OpenColorIOManagementSystem>;
+
+/// @class OpenColorIOManagementSystem
+/// Class for a default color management system.
+class MX_GENSHADER_API OpenColorIOManagementSystem : public DefaultColorManagementSystem
+{
+  public:
+    virtual ~OpenColorIOManagementSystem() { }
+
+    /// Create a new OpenColorIOManagementSystem
+    static OpenColorIOManagementSystemPtr create(const OCIO::ConstConfigRcPtr& config, const string& target);
```

Comment:
How is a `OCIO::ConstConfigRcPtr` constructed?  

This seems like the only instance where OCIO would be in the public MaterialX API, and it maybe be nice if could avoid that need. 

Perhaps if we exposed wrappers around `OCIO::Config::CreateFromEnv()`  `OCIO::Config::CreateFromFile()` `OCIO::Config::CreateFromBuiltinConfig()` ?